### DISCUSSION TITLE
New Requirement: The holding account should be able to control how many times the sub-account can be used per day, or how many times it can be used every hour.

### DISCUSSION THREAD

**Discussion by windkwbs** [open-webui/open-webui#3188] [General]
New Requirement: The holding account should be able to control how many times the sub-account can be used per day, or how many times it can be used every hour.

**Comment by pkeffect** [comment]
This can be done using the Pipelines system.

https://github.com/open-webui/pipelines/tree/main/examples/filters

There is an example of a rate limiter you can use to get started.

**Reply by windkwbs** [reply]
After modifying the [conversation_turn_limit_filter.py] file, it should be executed where in the open webui project?

**Reply by justinh-rahb** [reply]
No modifications should be necessary. It's to be used with the Pipelines server added to your WebUI as an "OpenAI" connection.

**Comment by pkeffect** [comment]
https://imgur.com/Z3fWnlm

Here in the Admin settings.

**Reply by windkwbs** [reply]
docker run -d -p 9099:9099 --add-host=host.docker.internal:host-gateway -v pipelines:/app/pipelines --name pipelines --restart always ghcr.io/open-webui/pipelines:main
Why doesn't the above command directly update the pipeline function? Because after using the command, there is actually an 863.4M file being downloaded in the command window.
